pkg/osquery/table/kolide_target_membership.go

```diff
@@ -0,0 +1,123 @@
+package table
+
+import (
+	"context"
+	"time"
+
+	"github.com/boltdb/bolt"
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/gogo/protobuf/proto"
+	"github.com/kolide/launcher/pkg/osquery"
+	"github.com/kolide/launcher/pkg/pb/kt"
+	"github.com/kolide/osquery-go/plugin/table"
+	"github.com/pkg/errors"
+)
+
+const targetMembershipKey = "target_membership"
+
+type TargetMembership struct {
+	logger       log.Logger
+	db           *bolt.DB
+	targetClient kt.KTargetClient
+}
+
+func NewTargetMembership(logger log.Logger, db *bolt.DB, targetClient kt.KTargetClient) TargetMembership {
+	return TargetMembership{
+		logger:       logger,
+		db:           db,
+		targetClient: targetClient,
+	}
+}
+
+func (tm *TargetMembership) Plugin() *table.Plugin {
+	columns := []table.ColumnDefinition{
+		table.TextColumn("id"),
+	}
+
+	return table.NewPlugin("kolide_target_membership", columns, tm.generate())
+}
+
+func (tm *TargetMembership) generate() table.GenerateFunc {
+	return func(ctx context.Context, queryContext table.QueryContext) ([]map[string]string, error) {
+
+		targets, err := tm.GetTargetMembership(ctx)
```

Comment:
Consider making this `tm.getTargetMembership` unless there is a reason to have this symbol exported.